rustls/src/client/ech.rs

```diff
@@ -360,6 +361,74 @@ impl EchState {
         )
     }
 
+    pub(crate) fn confirm_hrr_acceptance(
+        &self,
+        hrr: &HelloRetryRequest,
+        cs: &Tls13CipherSuite,
+        common: &mut CommonState,
+    ) -> Result<bool, Error> {
+        // The client checks for the "encrypted_client_hello" extension.
+        let ech_conf = match hrr.ech() {
+            // If none is found, the server has implicitly rejected ECH.
+            None => return Ok(false),
+            // Otherwise, if it has a length other than 8, the client aborts the
+            // handshake with a "decode_error" alert.
+            Some(ech_conf) if ech_conf.len() != 8 => {
+                return Err({
+                    common.send_fatal_alert(
+                        AlertDescription::DecodeError,
+                        PeerMisbehaved::IllegalHelloRetryRequestWithInvalidEch,
+                    )
+                })
+            }
+            Some(ech_conf) => ech_conf,
+        };
+
+        // Otherwise the client computes hrr_accept_confirmation as described in Section
+        // 7.2.1
+        let confirmation_transcript = self.inner_hello_transcript.clone();
+        let mut confirmation_transcript =
+            confirmation_transcript.start_hash(cs.common.hash_provider);
+        confirmation_transcript.rollup_for_hrr();
+        confirmation_transcript.add_message(&Self::hello_retry_request_conf(hrr));
+
+        let derived = server_ech_hrr_confirmation_secret(
+            cs.hkdf_provider,
+            &self.inner_hello_random.0,
+            confirmation_transcript.current_hash(),
+        );
+
+        Ok(
```

Comment:
(Same thing for the `Ok`-wrapping here.)

Reply:
Fixed.